anonymous_helpline_chatbot/core/conversations.py

```diff
@@ -1,97 +1,237 @@
+                       "FOR SHARE",
+                       (chat_id, chat_id))
+        return cursor.fetchone() or (None, None)
+
+    def get_conversing(self, chat_id: int) -> Conversing:
         """
         Get client and operator from a conversation with a user with the given identifier
 
-        Retrieves both telegram and local identifiers of both client and operator of the conversation, in which the
+        Retrieves messenger identifiers of both client and operator of the conversation, in which the
         given user takes part. This function can be used for understanding whether current user is a client or an
         operator in his conversation, retrieving information about his interlocutor, etc
 
-        :param tg_id: Telegram identifier of either a client or an operator
-        :return: If the given user is not in conversation, `((None, None), (None, None))` is returned. Otherwise a
-            tuple of two tuples is returned, where the first tuple describes the client, the second tuple describes the
-            operator, and each of them consists of two `int`s, the first of which is the telegram id of a person, the
-            second is the local id
+        Warning: if your application is working in a multi-threaded environment (which is extremely likely), the
+        conversations are expected to be started or finished sometimes, so something might change between the moment
+        when you call this function and the moment when you use the returned result. That's why it is highly recommended
+        to only use this function together with the `.lock_conversations_list` context manager. For example,
+
+        ```
+        # WRONG!!!
+        _, operator_chat_id = conversations_controller.get_conversing(some_user_chat_id)
+        if operator_chat_id is None:
+            # Another thread might have already created a conversation with that user, so we rely on outdated data
+            print("User is not in a conversation")  # Who knows???
+        ```
+
+        ```
+        # CORRECT
+        with conversations_controller.lock_conversations_list():
+            # While we're inside of this block, no conversations can start or finish. All the other threads will wait
+            _, operator_chat_id = conversations_controller.get_conversing(some_user_chat_id)
+            if operator_chat_id is None:
+                print("User is not in a conversation")
+                # Do something, relying on this fact
+            else:
+                print("User is in a conversation")
+                # Do something, relying on this fact
+
+            # If there was only the `else` part, and we didn't want to fixate the "not existing" state of a
+            # conversation, it would be better to use `conversations_controller.get_conversing_with_plocking` (see
+            # below)
+        ```
+
+        Also note that if you only need to do something if a conversation/request **does** exist for the given user,
```

Comment:
I think it's important to use a conjunction here to explicitly state that this works for both conversations and requests
```suggestion
        Also note that if you only need to do something if a conversation or request **does** exist for the given user,
```